File: handlebars/src/test/resources/issue307.js
Handlebars.registerHelper('dowork', function(context, options) {
  var ret = '', i, j;
  for (i = 0, j = context.length; i < j; i += 1) {
    if (context[i].name === 'pato') {
      ret += options.fn(context[i]);
    } 
  }
  return ret;
});

File: handlebars/src/test/java/com/github/jknack/handlebars/Issue307.java
package com.github.jknack.handlebars;

import java.io.File;
import java.io.IOException;

import org.junit.Test;

public class Issue307 extends AbstractTest {

  @Override
  protected void configure(final Handlebars handlebars) {
    try {
      handlebars.registerHelpers(new File("src/test/resources/issue307.js"));
    } catch (Exception ex) {
      throw new IllegalStateException(ex);
    }
  }

  @Test
  public void dowork() throws IOException {
    shouldCompileTo(
        "{{#dowork root/results}}name:{{name}}, age:{{age}}{{/dowork}}",
        $("root",
            $("results", new Object[]{$("name", "edgar", "age", 34), $("name", "pato", "age", 34) })),
        "name:pato, age:34");
  }

}
